### ISSUE TITLE
Live versions of queries

### ISSUE THREAD

**Issue by scullxbones** [scullxbones/akka-persistence-mongo#38]
* Support `NoRefresh` hint for existing queries `AllPersistenceIds` and `AllEvents`
* Missing `NoRefresh` should imply live, long-running queries

**Comment by thjaeckle** [comment]
Will the `NoRefresh` hint for `AllPersistenceId` also support a use case like "give me all currently known persistence ids"?
Currently I see that for every persistence id one MongoDb query is done:
```
command akka-persistence.$cmd command: distinct { distinct: "akka_persistence_journal", key: "pid", query: {} }
```
Which gets extremely slow when there are some thousand persistence ids :)

**Comment by scullxbones** [comment]
So no refresh is the mode we operate in today where the stream competes when all ids are read.  The opposite means that the query continues to supply results as new ids are added.

I don't expect the distinct query to be too slow, it should be an index scan of the top nodes of the btree.  Are you seeing slow performance?  If so, I will need to rethink.

**Comment by thjaeckle** [comment]
If I see correctly (from looking at the MongoDB console while following statement is invoked):
```
final ReadJournal readJournal =
  PersistenceQuery.get(context().system()).getReadJournalFor("akka-contrib-mongodb-persistence-readjournal");
final Source<String, BoxedUnit> thingIdsQuery =
  readJournal.query(AllPersistenceIds.getInstance(), NoRefresh.getInstance());

final List<String> ids = new ArrayList<>();
thingIdsQuery.runForeach(id -> ids.add(id), ActorMaterializer.create(context()));
```
A single "distinct" command is done for *each* persistenceId there is. I see the following message hundreds/thousands of times in the MongoDB console (because it takes >100ms each):
```
command akka-persistence.$cmd command: distinct { distinct: "akka_persistence_journal", key: "pid", query: {} }
```

**Comment by scullxbones** [comment]
Oh. Yeah that's bad.  I'll create a separate bug ticket for this one. Thanks for reporting!

**Comment by scullxbones** [comment]
Partially addressed by PR #69 - EventsByPersistenceId is now live.  Will be available in release `1.1.0`

**Comment by scullxbones** [comment]
Will be totally addressed by PR #71, expecting to release with 1.1.1

**Comment by scullxbones** [comment]
Things seem to be stabilized, marking this closed.
